packages/test/src/admin-utils.ts

```diff
@@ -192,8 +224,144 @@ export async function createCollectionContainingProduct(
     ],
     parentId,
   };
-  return await adminClient.query<Collection, MutationCreateCollectionArgs>(
-    CREATE_COLLECTION,
-    { input }
+  return (
+    (await adminClient.query<Collection, MutationCreateCollectionArgs>(
+      CREATE_COLLECTION,
+      { input }
+    )) as any
+  ).createCollection;
+}
+
+export async function createProduct(
+  adminClient: SimpleGraphQLClient,
+  translationFields: Omit<
+    ProductTranslationInput,
+    'customFields' | 'languageCode'
+  >
+) {
+  return (
+    (await adminClient.query<Product, MutationCreateProductArgs>(
+      CREATE_PRODUCT,
+      {
+        input: {
+          translations: [
+            {
+              languageCode: LanguageCode.En,
+              description: translationFields.description,
+              name: translationFields.name,
+              slug: translationFields.slug,
+            },
+          ],
+        },
+      }
+    )) as any
+  ).createProduct;
+}
+
+export async function assignOptionGroupsToProduct(
+  adminClient: SimpleGraphQLClient,
+  product: Product,
```

Comment:
We only need `product.id` to assign options, so let's only accept `productId` as input parameter

Reply:
I actually return the product after i updated it

Reply:
You do `newProduct = ( (await adminClient.query` before that right, so that means you can still return the new product, and only accept productId as input? Maybe I am misreading something

Reply:
You could do something like this I think? This way you can use `productId` as input instead of the whole `product`

```ts
let newproduct;
  for (const optionGroup of createdOptionGroups) {
    newProduct = (
      (await adminClient.query<Product, MutationAddOptionGroupToProductArgs>(
        ADD_OPTION_GROUP_TO_PRODUCT,
        {
          optionGroupId: optionGroup.id as string,
          productId: productId,
        }
      )) as any
    ).addOptionGroupToProduct;
  }
  return newProduct;
```